search-sentiments-map/public/scripts/map-script.js

```diff
@@ -174,37 +202,75 @@ function styleFeature(feature) {
 }
 
 /**
- * Responds to the mouse-in event on a map shape(country).
+ * Responds to the mouse-in event on a map shape (country or state).
  * @param {?google.maps.MouseEvent} e Mouse-in event.
  */
 function mouseInToRegion(e) {
-  let countryData = e.feature.getProperty('country_data');
-  // Add popup info window with country info.
-  if (countryData == null) {
+  let regionData = isWorldLevel ? 
+      e.feature.getProperty('country_data') : 
+      e.feature.getProperty('state_data');
+
+  if (regionData == null) {
     return;
   }
-  // Set the hover country so the `setStyle` function can change the
-  // border.
-  e.feature.setProperty('country', 'hover');
-  countryInfo = e.feature.getProperty('name') + ': ';
+  // Set the hover region so the `setStyle` function can change the border.
+  e.feature.setProperty('status', 'hover');
+  regionInfo = isWorldLevel ? 
+      e.feature.getProperty('name') + ': ' : 
+      e.feature.getProperty('NAME') + ': ';
 
   // Display "N/A" on hover when there are no results and thererfore the
   // sentiment score is the no results default score.
-  countryInfo +=
-      ((countryData === NO_RESULTS_DEFAULT_SCORE) ?
-          "N/A" : countryData.toLocaleString());
+  regionInfo +=
+      ((regionData === NO_RESULTS_DEFAULT_SCORE) ?
+          "N/A" : regionData.toLocaleString());
 
-  infowindow.setContent(countryInfo);
+  // Add popup info window with region info.
+  infowindow.setContent(regionInfo);
   infowindow.setPosition(e.latLng);
   infowindow.open(map);
 }
 
 /**
- * Responds to the mouse-out event on a map shape (country).
+ * Responds to the mouse-out event on a map shape (country or state).
  * @param {?google.maps.MouseEvent} e Mouse-out event.
  */
 function mouseOutOfRegion(e) {
-  // Reset the hover country, returning the border to normal. Close infowindow.
-  e.feature.setProperty('country', 'normal');
+  // Reset the hover status, returning the border to normal. Close infowindow.
+  e.feature.setProperty('status', 'normal');
   infowindow.close();
-}
\ No newline at end of file
+}
+
+/**
+ * Resets the map according to the zoom level (US or world) selected by the 
+ * user by adjusting the map center, zoom level, polygons, and displayed data.
+ * TODO(chenyuz): 
+ * 1. On US level, add button to allow switching to US trends.
+ * 2. Toggle the display of mode selector. Show on hover of the select that 
+ * only popularity is available.
+ * 3. Modify user search to be for popularity only.
+ */
+function resetMapZoomLevel() {
+  const zoomLevel = document.getElementById('zoom-level-select').value;
+  // Clear the previous map features.
+  map.data.forEach(function(feature) {
+    map.data.remove(feature);
+  });
+
+  if (zoomLevel === 'us') {
+    map.setCenter(US_CENTER_COORDINATES);
+    map.setZoom(US_ZOOM_LEVEL);
+    isWorldLevel = false;
+    isSentimentMode = false;  // Sentiment mode is not available at US level.
```

Comment:
Should you set it to true in the else block?

Reply:
Actually no, because isSentimentMode gets reset in `loadRegionDataByMode`. The real bug is that I should move the reset into the if block